src/server/run_sched.c

```diff
@@ -155,6 +159,132 @@ put_sched_cmd(int sock, int cmd, char *jobid)
 	return -1;
 }
 
+/**
+ * @brief
+ * 		find_assoc_sched - find the corresponding scheduler which is responsible
+ * 		for handling this job.
+ *
+ * @param[in]	jid - job id
+ * @param[out]	target_sched - pointer to the corresponding scheduler to which the job belongs to
+ *
+ * @retval - 1  if success
+ * 	   - 0 if fail
+ */
+int
+find_assoc_sched_jid(char *jid, pbs_sched **target_sched)
+{
+	job *pj;
+	pbs_queue *pq;
+	char *q_name;
+	pbs_sched *psched;
+	int t;
+
+	*target_sched = NULL;
+
+	t = is_job_array(jid);
+	if ((t == IS_ARRAY_NO) || (t == IS_ARRAY_ArrayJob))
+		pj = find_job(jid);		/* regular or ArrayJob itself */
+	else
+		pj = find_arrayparent(jid); /* subjob(s) */
+
+	if (pj == NULL)
+		return 0;
+
+	q_name = pj->ji_qs.ji_queue;
```

Comment:
at this point why don't you call find_assoc_sched_pque? code is almost same in both the functions

Reply:
Implemented.